error while running "make -j$(nproc)"

## Build Process
*Describe how you build polybar, list the exact commands you are using:*

> git clone --recursive https://github.com/polybar/polybar
> cd polybar
> mkdir build
> cd build
> cmake ..
> make -j$(nproc)

**If you build polybar directly from this repository:**

* Output of `git describe --tags`:

3.4.0-113-ga625e2b

**If you use some other way (like the AUR):**

*List exactly where you are building from and which version you are building.*

~/notes/polybar/polybar/build
polybar version - 3.4.0


## Build Log
*Post everything that is output to the terminal while building polybar below. This HAS to include the output of the `cmake` and `make` commands, if you used them.*

```
❯ cmake ..
-- Trying to enable ccache
-- Couldn't locate ccache, disabling ccache...
-- The CXX compiler identification is GNU 7.2.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- No build type specified; using Release
-- Using supported compiler GNU-7.2.0
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1")
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE
-- Checking for module 'cairo-fc'
--   Found cairo-fc, version 1.14.10
-- Checking for module 'alsa'
--   Found alsa, version 1.1.3
-- Checking for module 'libcurl'
--   Found libcurl, version 7.55.1
-- Checking for module 'libmpdclient'
--   Found libmpdclient, version 2.9
-- Checking for module 'libnl-genl-3.0'
--   Found libnl-genl-3.0, version 3.2.29
-- Checking for module 'libpulse'
--   Found libpulse, version 10.0
-- Checking for module 'xcb-composite'
--   Found xcb-composite, version 1.12
-- Checking for module 'xcb-xkb'
--   Found xcb-xkb, version 1.12
-- Checking for module 'xcb-randr'
--   Found xcb-randr, version 1.12
-- Checking for module 'xcb-randr>=1.12'
--   Found xcb-randr, version 1.12
-- Checking for module 'xcb-xrm'
--   Found xcb-xrm, version 1.0
-- Checking for module 'xcb-cursor'
--   Found xcb-cursor, version 0.1.1
--  Build:
--    Version: 3.4.0-113-ga625e2b
--    Type: Release
--    CXX: /usr/bin/c++  -Wall -Wextra -Wpedantic -O3 -DNDEBUG
--    LD: /usr/bin/ld
--  Install Paths:
--    PREFIX:  /usr/local
--    BINDIR:  /usr/local/bin
--    DATADIR: /usr/local/share
--    DOCDIR:  /usr/local/share/doc/polybar
--    MANDIR:  /usr/local/share/man
--  Targets:
-- [X]   polybar-msg
-- [ ]   testsuite
-- [ ]   documentation
--  Module support:
-- [X]   alsa (1.1.3)
-- [X]   curl (7.55.1)
-- [X]   i3
-- [X]   mpd (2.9)
-- [X]   network (libnl) (3.2.29)
-- [X]   pulseaudio (10.0)
-- [X]   xkeyboard (1.12)
--  X extensions:
-- [X]   xcb-randr (1.12)
-- [X]   xcb-randr (monitor support) (1.12)
-- [X]   xcb-composite (1.12)
-- [X]   xcb-xkb (1.12)
-- [X]   xcb-xrm (1.0)
-- [X]   xcb-cursor (0.1.1)
--  Log options:
-- [X]   Trace logging
-- The C compiler identification is GNU 7.2.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Checking for module 'xcb-proto'
--   Found xcb-proto, version 1.12
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.6.3", minimum required is "3.5")
-- XCB[XCB]: Found component XCB
-- Found XCB_XCB: /usr/lib/x86_64-linux-gnu/libxcb.so
-- XCB[ICCCM]: Found component ICCCM
-- Found XCB_ICCCM: /usr/lib/x86_64-linux-gnu/libxcb-icccm.so
-- XCB[EWMH]: Found component EWMH
-- Found XCB_EWMH: /usr/lib/x86_64-linux-gnu/libxcb-ewmh.so
-- XCB[UTIL]: Found component UTIL
-- Found XCB_UTIL: /usr/lib/x86_64-linux-gnu/libxcb-util.so
-- XCB[IMAGE]: Found component IMAGE
-- Found XCB_IMAGE: /usr/lib/x86_64-linux-gnu/libxcb-image.so
-- Found XCB: /usr/lib/x86_64-linux-gnu/libxcb.so;/usr/lib/x86_64-linux-gnu/libxcb-icccm.so;/usr/lib/x86_64-linux-gnu/libxcb-ewmh.so;/usr/lib/x86_64-linux-gnu/libxcb-util.so;/usr/lib/x86_64-linux-gnu/libxcb-image.so
-- Searching for xcbgen with /usr/bin/python3
-- Searching for xcbgen with python3
-- Searching for xcbgen with python
-- Found xcbgen in /usr/lib/python2.7/dist-packages
-- XCB[RANDR]: Found component RANDR
-- Found XCB_RANDR: /usr/lib/x86_64-linux-gnu/libxcb-randr.so
-- Found XCB: /usr/lib/x86_64-linux-gnu/libxcb.so;/usr/lib/x86_64-linux-gnu/libxcb-icccm.so;/usr/lib/x86_64-linux-gnu/libxcb-ewmh.so;/usr/lib/x86_64-linux-gnu/libxcb-util.so;/usr/lib/x86_64-linux-gnu/libxcb-image.so;/usr/lib/x86_64-linux-gnu/libxcb-randr.so
-- XCB[COMPOSITE]: Found component COMPOSITE
-- Found XCB_COMPOSITE: /usr/lib/x86_64-linux-gnu/libxcb-composite.so
-- Found XCB: /usr/lib/x86_64-linux-gnu/libxcb.so;/usr/lib/x86_64-linux-gnu/libxcb-icccm.so;/usr/lib/x86_64-linux-gnu/libxcb-ewmh.so;/usr/lib/x86_64-linux-gnu/libxcb-util.so;/usr/lib/x86_64-linux-gnu/libxcb-image.so;/usr/lib/x86_64-linux-gnu/libxcb-randr.so;/usr/lib/x86_64-linux-gnu/libxcb-composite.so
-- XCB[XKB]: Found component XKB
-- Found XCB_XKB: /usr/lib/x86_64-linux-gnu/libxcb-xkb.so
-- Found XCB: /usr/lib/x86_64-linux-gnu/libxcb.so;/usr/lib/x86_64-linux-gnu/libxcb-icccm.so;/usr/lib/x86_64-linux-gnu/libxcb-ewmh.so;/usr/lib/x86_64-linux-gnu/libxcb-util.so;/usr/lib/x86_64-linux-gnu/libxcb-image.so;/usr/lib/x86_64-linux-gnu/libxcb-randr.so;/usr/lib/x86_64-linux-gnu/libxcb-composite.so;/usr/lib/x86_64-linux-gnu/libxcb-xkb.so
-- xpp: including xcb proto composite.xml
-- xpp: including xcb proto randr.xml
-- xpp: including xcb proto xkb.xml
-- xpp: including xcb proto xproto.xml
-- Checking for module 'jsoncpp'
--   Found jsoncpp, version 1.7.4
-- Font not found: fixed:pixelsize=10
-- Font not found: unifont:fontformat=truetype
-- Font not found: siji:pixelsize=10
-- Configuring done
-- Generating done
-- Build files have been written to: /home/joker70/notes/polybar/polybar/build
```

```
❯ make -j$(nproc)
[  1%] Generating generated-sources/include/xpp/proto/composite.hpp
Scanning dependencies of target i3ipc++
Scanning dependencies of target polybar-msg
[  2%] Generating generated-sources/include/xpp/proto/x.hpp
[  3%] Building CXX object bin/CMakeFiles/polybar-msg.dir/ipc.cpp.o
[  4%] Building CXX object lib/i3ipcpp/CMakeFiles/i3ipc++.dir/src/ipc.cpp.o
Traceback (most recent call last):
  File "/home/joker70/notes/polybar/polybar/lib/xpp/generators/cpp_client.py", line 3096, in <module>
    from xcbgen.state import Module
  File "/usr/lib/python2.7/dist-packages/xcbgen/state.py", line 7, in <module>
    from xcbgen import matcher
  File "/usr/lib/python2.7/dist-packages/xcbgen/matcher.py", line 12, in <module>
    from xcbgen.xtypes import *
  File "/usr/lib/python2.7/dist-packages/xcbgen/xtypes.py", line 504
    print "Explicit start-align for %s: %s\n" % (self, self.required_start_align)
                                            ^
SyntaxError: invalid syntax
Traceback (most recent call last):
  File "/home/joker70/notes/polybar/polybar/lib/xpp/generators/cpp_client.py", line 3096, in <module>
    from xcbgen.state import Module
  File "/usr/lib/python2.7/dist-packages/xcbgen/state.py", line 7, in <module>
    from xcbgen import matcher
  File "/usr/lib/python2.7/dist-packages/xcbgen/matcher.py", line 12, in <module>
    from xcbgen.xtypes import *
  File "/usr/lib/python2.7/dist-packages/xcbgen/xtypes.py", line 504
    print "Explicit start-align for %s: %s\n" % (self, self.required_start_align)
                                            ^
SyntaxError: invalid syntax
lib/xpp/CMakeFiles/xpp.dir/build.make:61: recipe for target 'lib/xpp/generated-sources/include/xpp/proto/composite.hpp' failed
make[2]: *** [lib/xpp/generated-sources/include/xpp/proto/composite.hpp] Error 1
make[2]: *** Deleting file 'lib/xpp/generated-sources/include/xpp/proto/composite.hpp'
make[2]: *** Waiting for unfinished jobs....
lib/xpp/CMakeFiles/xpp.dir/build.make:73: recipe for target 'lib/xpp/generated-sources/include/xpp/proto/x.hpp' failed
make[2]: *** [lib/xpp/generated-sources/include/xpp/proto/x.hpp] Error 1
make[2]: *** Deleting file 'lib/xpp/generated-sources/include/xpp/proto/x.hpp'
CMakeFiles/Makefile2:384: recipe for target 'lib/xpp/CMakeFiles/xpp.dir/all' failed
make[1]: *** [lib/xpp/CMakeFiles/xpp.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  5%] Building CXX object lib/i3ipcpp/CMakeFiles/i3ipc++.dir/src/ipc-util.cpp.o
[  6%] Building CXX object bin/CMakeFiles/polybar-msg.dir/utils/env.cpp.o
[  7%] Building CXX object bin/CMakeFiles/polybar-msg.dir/utils/file.cpp.o
[  8%] Building CXX object bin/CMakeFiles/polybar-msg.dir/utils/string.cpp.o
[  9%] Linking CXX executable polybar-msg
[  9%] Built target polybar-msg
[ 10%] Linking CXX static library libi3ipc++.a
[ 10%] Built target i3ipc++
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

## Environment:
* Distro (with Version):
Ubuntu - 17.10

## Additional context
*Add any other context that you think is necessary about the problem here.*

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,7 +24,7 @@ else()
     pkg_get_variable(XCBPROTO_XCBINCLUDEDIR xcb-proto xcbincludedir)
 endif()
 
-find_package(PythonInterp 3.5 REQUIRED)
+find_package(PythonInterp 2.7 REQUIRED)
 find_package(XCB REQUIRED XCB ICCCM EWMH UTIL IMAGE)
 
 if(NOT PYTHON_EXECUTABLE)
@@ -51,8 +51,8 @@ set(XPP_LIBRARIES
 #
 # Loop through a hardcoded list of python executables to locate the python module "xcbgen"
 #
-# TODO drop python2 once ubuntu and debian ship python3-xcbgen in their
-# maintained distros
+# TODO drop python2 once ubuntu and debian ship python3-xcbgen in all of their
+# maintained distros. This also means reintroducing FindPythonInterp 3.5
 foreach(CURRENT_EXECUTABLE ${PYTHON_EXECUTABLE} python3 python python2 python2.7)
   message(STATUS "Searching for xcbgen with " ${CURRENT_EXECUTABLE})
 
@@ -65,7 +65,8 @@ foreach(CURRENT_EXECUTABLE ${PYTHON_EXECUTABLE} python3 python python2 python2.7
   # When a shell script returns successfully its return code is 0
   if(_xcbgen_status EQUAL 0)
     set(PYTHON_XCBGEN "${_xcbgen_location}" CACHE STRING "Location of python module: xcbgen ")
-    message(STATUS "Found xcbgen in " ${PYTHON_XCBGEN})
+    set(PYTHON_EXECUTABLE "${CURRENT_EXECUTABLE}")
+    message(STATUS "Found xcbgen in ${PYTHON_XCBGEN} with ${PYTHON_EXECUTABLE}")
     break()
   endif()
 

